packages/views.api.unstable/src/utils/asyncValue.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface AsyncNotStartedLoadingValue {
+  type: "not-started";
+}
+
+export interface AsyncLoadingValue {
+  type: "loading";
+}
+
+export interface AsyncLoadedValue<V> {
+  type: "loaded";
+  value: V | undefined;
```

Comment:
I still want to spend more time thinking about the `| undefined`s. If we support marking parameters as required we could treat undefined as a failed state, or a default value. One for later